services/QuillLMS/app/services/analytics/segment_analytics.rb

```diff
@@ -21,38 +21,35 @@ def track_event_from_string(event_name, user_id)
     # make sure that event name is written as a string in the pattern of
     # those in app/services/analytics/segment_io.rb
     # i.e. "BUILD_YOUR_OWN_ACTIVITY_PACK"
-    user = User.find(user_id)
-    track(user, {
+    track({
        user_id: user_id,
        event: "SegmentIo::BackgroundEvents::#{event_name}".constantize
       })
   end
 
   def track_activity_assignment(teacher_id, activity_id)
-    user = User.find(teacher_id)
     activity = Activity.find(activity_id)
```

Comment:
Good call. Doing traces, I didn't find occasions of this method being passed `uid`, but I agree that it makes sense to make them consistent. Will update.

Reply:
There is a `.find_by_id_or_uid` method on the activity model you could use!

Reply:
Although as I'm doing the Rails 5 upgrade, I'm realizing that instead would recommend a non-dynamic finder with: : `Activity.find_by(id: activity_id) || Activity.find_by(uid: activity_id)`

Reply:
@emilia-friedberg Good point. I was going to use that method, but `Activity.find_by_id_or_uid` raises if a record isn't found, which I was trying to avoid here, so I went with the non-raising finders.